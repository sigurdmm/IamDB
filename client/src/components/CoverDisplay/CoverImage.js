import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import defaultImage from '../InformationSection/no-image-found.jpg';

import './CoverImage.less';

/**
 * Return an em tag with rating if rating is passed as prop to CoverImage
 */
const getRating = (rating = null) => {
  if (!rating) {
    return (null);
  }
  return <em className='cover__overlay__rating'>⭐️: {rating}/10</em>;
};

/**
 * Return an strong tag with the a title
 * If released is passed as a prop to CoverImage it will return title(released)
 */
const getTitleAndYear = (title, released = null) => {
  // If no release year is passed, don't render release year
  if (!released) {
    return <strong className='cover__overlay__title'>{title}</strong>;
  }
  // else render "title (year)"
  const timestamp = new Date(parseInt(released, 10));
  const year = timestamp.getFullYear();
  return <strong className='cover__overlay__title'>{`${title} (${year})`}</strong>;
};
/**
 * Renders a div with a thumbnail. If no thumbnail is passed as a prop
 * it renders a default image.
 */
const CoverImage = ({
  thumbnail,
  title,
  rating,
  released,
  url,
}) => <div className={`cover ${!thumbnail ? 'cover--inverse' : ''}`}>
  <Link to={url}>
    <img className='cover__thumbnail' src={thumbnail || defaultImage}/>
    <div className='cover__overlay'>
      {getTitleAndYear(title, released)}
      {getRating(rating)}
    </div>
  </Link>
</div>;

CoverImage.propTypes = {
  thumbnail: PropTypes.string,
  title: PropTypes.string.isRequired,
  rating: PropTypes.number,
  url: PropTypes.string,
  released: PropTypes.any,
};

export default CoverImage;
