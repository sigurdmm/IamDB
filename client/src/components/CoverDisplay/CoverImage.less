@import '../../constants';

.cover {
  max-height: 200px;
  width: 120px;
  height: 400px;
  position: relative;
  display: inline-block;
  overflow: hidden;
  margin: 0;

  &__thumbnail {
    display: block;
    max-width: 120px;
    margin-left: auto;
    margin-right: auto;
    opacity: 0.3;

    position: absolute;

    top: 50%;
    left: 50%;

    min-height: 100%;
    min-width: auto;
    transform: translate(-50%, -50%);
  }

  img:hover {
    opacity: 1.0;
  }

  &__overlay {
    padding: 4px;
    font-weight: bold;
    color: @textColor;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    height: 100%;

    &__title {

    }

    &__rating {
    }
  }
}
