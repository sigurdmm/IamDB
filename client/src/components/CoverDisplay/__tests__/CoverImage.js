import { render } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import CoverImage from '../CoverImage';

describe('<CoverImage/>', () => {
  it('should render img, when link exists', () => {
    const wrapper = render(<MemoryRouter>
      <CoverImage rating={5} title={'title'} thumbnail="https://example.com/test.jpg" url='url'/>
    </MemoryRouter>);

    expect(wrapper).toMatchSnapshot();
  });
  it('should render img, when link does not exist', () => {
    const wrapper = render(<MemoryRouter>
      <CoverImage rating={3} title={'title'} thumbnail={null} url='url'/>
    </MemoryRouter>);

    expect(wrapper).toMatchSnapshot();
  });
});
