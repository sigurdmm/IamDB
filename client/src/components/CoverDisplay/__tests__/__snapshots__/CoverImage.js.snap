// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CoverImage/> should render img, when link does not exist 1`] = `
<div
  class="cover cover--inverse"
>
  <a
    href="/url"
  >
    <img
      class="cover__thumbnail"
      src="no-image-found.jpg"
    />
    <div
      class="cover__overlay"
    >
      <strong
        class="cover__overlay__title"
      >
        title
      </strong>
      <em
        class="cover__overlay__rating"
      >
        ⭐️: 3/10
      </em>
    </div>
  </a>
</div>
`;

exports[`<CoverImage/> should render img, when link exists 1`] = `
<div
  class="cover "
>
  <a
    href="/url"
  >
    <img
      class="cover__thumbnail"
      src="https://example.com/test.jpg"
    />
    <div
      class="cover__overlay"
    >
      <strong
        class="cover__overlay__title"
      >
        title
      </strong>
      <em
        class="cover__overlay__rating"
      >
        ⭐️: 5/10
      </em>
    </div>
  </a>
</div>
`;
